### PR TITLE
Hotfix

### PR DESCRIPTION
**Pull request by Changron**
1. fix bug created in #129, can create contest now
2. refactor redirect
3. only subjudge, judge, admin can see register button
4. make `get_owned_group` in `group/group_info.py` and `get_contests` in `contest_archive.py` distinct

### CODE REVIEW

File: contest/views.py [hunk 1]
```diff
@@ -113,11 +113,11 @@ def contest(request, cid):
...
 @login_required
 def new(request):
+    title = "New Contest"
     if can_create_contest(request.user):
         if request.method == 'GET':
             form = ContestForm(initial=\
-                {'user':request.user, 'method':request.method})
-            title = "New Contest"
+                {'owner':request.user, 'user':request.user, 'method':request.method})
             return render_index(request,'contest/editContest.html',{'form':form,'title':title})
```
Comment (nickelcheng) [inline contest/views.py]: This line is too long.

File: contest/views.py [hunk 2]
```diff
@@ -126,6 +126,8 @@ def new(request):
                 logger.info('Contest: User %s Create a new contest %s!' %
                     (request.user ,new_contest.id))
                 return redirect('contest:archive')
+            else:
+                return render_index(request,'contest/editContest.html',{'form':form,'title':title})
```
Comment (nickelcheng) [inline contest/views.py]: This line is too long.

File: contest/views.py [hunk 3]
```diff
@@ -152,8 +154,7 @@ def edit(request, cid):
                     (request.user, modified_contest.id))
                 return redirect('contest:archive')
             else:
-                return render_index(request,'contest/editContest.html',
-                    {'form':form,'user':request.user,'title':title})
+                return render_index(request,'contest/editContest.html',{'form':form,'title':title})
```
Comment (nickelcheng) [inline contest/views.py]: This line is too long.

**Comment by Changron** [comment]
fixed

**Comment by nickelcheng** [comment]
+1
@CHIAHSINHOU please test

**Comment by CHIAHSINHOU** [comment]
Because there is some bugs, I can't test this PR. Henry said he would fix it later.

**Comment by henryyang42** [comment]
You can still test the function if you relaunch the server.

**Comment by CHIAHSINHOU** [comment]
This PR is ok for me. However, I think there is still something we can improve such that the size of row when it shows the contests in Contest Archive. The size of row in first page is different from the size in other pages.

**Comment by Changron** [comment]
since the width is not enough
so the row automatically become higher

**Comment by nickelcheng** [comment]
@bruce3557 please check

File: contest/static/contest/js/editContest.js [hunk 4]
```diff
@@ -20,6 +20,7 @@ SOFTWARE.
 $(document).ready(function() {
     add_attribute('id_coowner', 'class', 'searchable');
     add_attribute('id_problem', 'class', 'searchable');
+    hide('id_owner');
```
Comment (bruce3557) [inline contest/static/contest/js/editContest.js]: In fact, I'm curious about whether we can generalize the ```hide``` function with class or id level.
That will be great but fine with this PR!

**Comment by bruce3557** [comment]
+1 after fixing comments

**Comment by nickelcheng** [comment]
About @CHIAHSINHOU's comment, can you fix length of the row so that it can accommodate 50 characters (in our schema, cname's max length = 50)

**Comment by Changron** [comment]
OK for number
but not sure about chinese characters
and it depends on the window size user has.

**Comment by Changron** [comment]
@bruce3557 
what do you mean?

**Comment by nickelcheng** [comment]
going to merge

**Pull request merged**